### PR TITLE
test: Add collective scheduling tests

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- Breaks down tests into separate files as single file was getting bigger and bigger causing greater chances of conflicts and making it difficult to find a test.
- Adds Collective Scheduling test
- Improves logging where I felt it was needed during debugging of tests
image

## How should this be tested?

`yarn test team-bookings`

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/lib/piiFreeData.ts [hunk 1]
```diff
@@ -3,6 +3,14 @@ import type { Credential, SelectedCalendar, DestinationCalendar } from "@prisma/
 import type { EventType } from "@calcom/prisma/client";
 import type { CalendarEvent } from "@calcom/types/Calendar";
 
+function getBooleanStatus<T extends unknown>(val: T) {
+  if (process.env.NODE_ENV === "production") {
+    return `PiiFree:${!!val}`;
```
Comment (hariombalhara) [inline packages/lib/piiFreeData.ts:8]: Mark that the data has been redacted,

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 2]
```diff
@@ -968,7 +967,7 @@ async function handler(
     if (
       availableUsers.filter((user) => user.isFixed).length !== users.filter((user) => user.isFixed).length
     ) {
-      throw new Error("Some users are unavailable for booking.");
+      throw new Error("Some of the hosts are unavailable for booking.");
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:970]: I think this is a more apt error

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 3]
```diff
@@ -379,7 +379,6 @@ async function ensureAvailableUsers(
       )
     : undefined;
 
-  log.debug("getUserAvailability for users", JSON.stringify({ users: eventType.users.map((u) => u.id) }));
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:382]: Moved this log to `getUserAvailability`

File: apps/web/test/utils/bookingScenario/bookingScenario.ts [hunk 4]
```diff
@@ -52,24 +50,29 @@ type ScenarioData = {
...
   schedules: {
-    id: number;
+    // Allows giving id in the input directly so that it can be referenced somewhere else as well
+    id?: number;
     name: string;
     availability: {
-      userId: number | null;
-      eventTypeId: number | null;
+      // userId: number | null;
+      // eventTypeId: number | null;
```
Comment (zomars) [inline apps/web/test/utils/bookingScenario/bookingScenario.ts:75]: Can we remove instead?

File: apps/web/test/utils/bookingScenario/bookingScenario.ts [hunk 5]
```diff
@@ -197,10 +234,24 @@ async function addEventTypes(eventTypes: InputEventType[], usersStore: InputUser
...
+              },
+            },
+          }
+        : eventType.schedule,
     };
   });
   log.silly("TestData: Creating EventType", JSON.stringify(eventTypesWithUsers));
-  await addEventTypesToDb(eventTypesWithUsers);
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
```
Comment (zomars) [inline apps/web/test/utils/bookingScenario/bookingScenario.ts:253]:
```suggestion
  // @ts-expect-error Put a reason on why here
```

File: apps/web/test/utils/bookingScenario/bookingScenario.ts [hunk 6]
```diff
@@ -497,21 +587,37 @@ export const TestData = {
...
+    },
     IstWorkHoursWithDateOverride: (dateString: string) => ({
-      id: 1,
       name: "9:30AM to 6PM in India - 4:00AM to 12:30PM in GMT but with a Date Override for 2PM to 6PM IST(in GST time it is 8:30AM to 12:30PM)",
       availability: [
         {
-          userId: null,
-          eventTypeId: null,
+          // userId: null,
+          // eventTypeId: null,
```
Comment (zomars) [inline apps/web/test/utils/bookingScenario/bookingScenario.ts:612]: Can we remove instead? Or are we planning to uncomment in the future?

File: apps/web/test/utils/bookingScenario/expects.ts [hunk 7]
```diff
@@ -182,11 +182,15 @@ export function expectSuccessfulBookingCreationEmails({
   emails,
   organizer,
   booker,
+  guests,
```
Comment (zomars) [inline apps/web/test/utils/bookingScenario/expects.ts:185]: guests seems to be unused?

**Comment by hariombalhara** [comment]
Thanks for review @zomars 🙏 Addressed the feedback.

File: apps/web/test/utils/bookingScenario/bookingScenario.ts (hunk 5 above)
Comment (hariombalhara) [reply apps/web/test/utils/bookingScenario/bookingScenario.ts]: Removed ts-ignore

File: apps/web/test/utils/bookingScenario/bookingScenario.ts (hunk 6 above)
Comment (hariombalhara) [reply apps/web/test/utils/bookingScenario/bookingScenario.ts]: Done. I think it won't be needed.

File: apps/web/test/utils/bookingScenario/expects.ts (hunk 7 above)
Comment (hariombalhara) [reply apps/web/test/utils/bookingScenario/expects.ts:185]: Started using it. Plan was to add the verification for guest emails as well. Have added the support in the function, though guests isn't being passed as of now.

**Review by zomars — APPROVED** [review]
Looks good to merge now 🙏🏽

**Pull request merged**